docs/proposals/permissive-mtl.md

```diff
@@ -0,0 +1,42 @@
+# Permissive mTLS mode
+
+## Context 
+
+The goal for this proposal is to provide a more pragmatic way to enable mTLS in a service mesh. When working with existing applications, 
+it is not practical to have mTLS enabled for everything or nothing, but we need to provide a more gradual way to do it.
+Therefore, a "permissive" mTLS mode (as opposed to the "strict" one we use today) would allow a team to enable mTLS yet allow 
+traffic that is not part of the zero-trust infrastructure to still be able to make requests without a strict validation of the 
+certificates on incoming requests.
+
+## Configuration
+
+Add a new field `mode` under the `mtls.backends` section. The type of the field is enum with 2 values: `strict` and `permissive`.
+To keep backwards compatibility a default value should be `strict`.
+
+```yaml
+apiVersion: kuma.io/v1alpha1
+kind: Mesh
+metadata:
+  name: default
+spec:
+  mtls:
+    enabledBackend: ca-1
+    backends:
+      - name: ca-1
+        type: builtin
+        mode: strict|permissive
```

Comment:
Could we instead just flag it as `permissive: true`? As the default would already be correct and we'd keep `mode` for the day it makes sense to have a mtls mode.

Reply:
Are we going to be supporting more than one mode? In the future do we foresee having something like `zone-permissive` where only one zone is permissive?

If we plant to expand to other modes, then we need a more flexible way of configuring those modes. If not, `permissive: true` may work.

IMHO, we should optimize for flexibility.

Reply:
+1 for flexibility. I'd rather have an enum even if we see only two modes right now

Reply:
While I understand the motivation of flexibility. Wouldn't any kind of extra permissiveness actually require more information so that an enum by itself wouldn't be sufficient?
In the example of a `zone-permissive` you'd probably want to provide a list of zone which are `permissive` or not. So in the end it's `permissive: true` and some extra config.
Another option you might want to add is `permissive-mesh` where you'd want to specify meshes that are allowed to talk to us which would also require extra config.

In any case I'm just suggesting and if you think an enum is better I'm good with this too.